streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestUtil.java

```diff
@@ -45,21 +47,48 @@
             public Processor<Object, Object> get() {
                 return new AbstractProcessor<Object, Object>() {
                     private int numRecordsProcessed = 0;
+                    private long smallestOffset = Long.MAX_VALUE;
+                    private long largestOffset = Long.MIN_VALUE;
 
                     @Override
                     public void init(final ProcessorContext context) {
                         super.init(context);
+                        LOG.info("[DEV] initializing processor: topic=" + topic + " taskId=" + context.taskId());
                         System.out.println("[DEV] initializing processor: topic=" + topic + " taskId=" + context.taskId());
+                        System.out.flush();
                         numRecordsProcessed = 0;
+                        smallestOffset = Long.MAX_VALUE;
+                        largestOffset = Long.MIN_VALUE;
                     }
 
                     @Override
                     public void process(final Object key, final Object value) {
                         numRecordsProcessed++;
-                        if (numRecordsProcessed % 100 == 0) {
```

Comment:
I had to take this out to make the upgrade test pass. The problem was that after upgrading, not all the instances would get a task for the input topic (`data`). When an instance has only to process repartition topics, it doesn't have very high throughput, and wouldn't even see 100 records in a partition before the test timed out and failed, since it never saw the "processed" message in stdout.

This is mildly concerning, but since this is a new mode of operation for this test, and since people have been complaining for a while about skewed assignments in 2.5 and before, I think we can just accept it in older branches.

To clarify, by removing this condition, we print to stdout on every record we process, so when an instance only has the `min` repartition topic, for example, it'll still print something, even if it only gets two records to process within the time limit.